node-src/git/git.ts

```diff
@@ -90,6 +90,29 @@ export async function getBranch() {
   }
 }
 
+// Retrieve the hash of all uncommitted files, which includes staged, unstaged, and untracked files,
+// excluding deleted files (which can't be hashed) and ignored files. There is no one single Git
+// command to reliably get this information, so we use a combination of commands grouped together.
+export async function getUncommittedHash() {
+  const listStagedFiles = 'git diff --name-only --diff-filter=d --cached';
+  const listUnstagedFiles = 'git diff --name-only --diff-filter=d';
+  const listUntrackedFiles = 'git ls-files --others --exclude-standard';
+  const listUncommittedFiles = [listStagedFiles, listUnstagedFiles, listUntrackedFiles].join(';');
+
+  const uncommittedHash = (
+    await execGitCommand(
+      // Pass the combined list of filenames to hash-object to retrieve a list of hashes. Then pass
+      // the list of hashes to hash-object again to retrieve a single hash of all hashes. We use
+      // stdin to avoid the limit on command line arguments.
+      `(${listUncommittedFiles}) | git hash-object --stdin-paths | git hash-object --stdin`
```

Comment:
As I mentioned, I don't think `x ; y ; z` does what you think it does in bash. Am I wrong?

Reply:
I tried it and it gave me a concatenated list of the results of the three operations. So long as they're wrapped in parenthesis, the piping works. But maybe I missed something.